python/pyspark/sql/session.py

```diff
@@ -478,13 +478,22 @@ def _inferSchema(self, rdd, samplingRatio=None, names=None):
             schema = rdd.map(lambda row: _infer_schema(row, names)).reduce(_merge_type)
         return schema
 
-    def _createFromRDD(self, rdd, schema, samplingRatio):
+    def _create_verified_converter(self, schema, verifySchema):
+        converter = _create_converter(schema)
+        verify_func = _make_type_verifier(schema) if verifySchema else lambda _: True
+
+        def verify_and_convert(obj):
+            verify_func(obj)
```

Comment:
Hm, why do we need to verify this again? The verification is done at `prepare(data)`

Reply:
I still think we should avoid doing this again though. the verification func is pretty expensive

Reply:
https://github.com/apache/spark/blob/132cbf0c8c1a382f33d8d212f931f5956f85a2f9/python/pyspark/sql/session.py#L695

It is not verified twice, previously, it is not verified. See the above `identity` prepare code snippet.

Reply:
In the latest commit, I've introduce a flag and avoid invoking the identity `prepare` function.